src/Rings.js

```diff
@@ -0,0 +1,728 @@
+        const node = edge.source.id === p.id ? edge.target : edge.source;
+        const s = radian / total;
+        const a = angle - s * children / 2 + s / 2 + s * i;
+
+        node.radians = a;
+        node.x = width / 2 + secondaryRing * Math.cos(a);
+        node.y = height / 2 + secondaryRing * Math.sin(a);
+        
+        secondaries.push(node);
+      });
+    });
+
+    const primaryDistance = ringWidth / 2;
+    const secondaryDistance = ringWidth / 4;
+
+    let primaryMax = primaryDistance / 2 - 4;
+    if (primaryDistance / 2 - 4 < 8) {
+      primaryMax = min([primaryDistance / 2, 8]);
+    }
+
+    let secondaryMax = secondaryDistance / 2 - 4;
+    if (secondaryDistance / 2 - 4 < 4) {
+      secondaryMax = min([secondaryDistance / 2, 4]);
+    }
+
+    if (secondaryMax > ringWidth / 10) {
+      secondaryMax = ringWidth / 10;
+    }
+
+    if (secondaryMax > primaryMax && secondaryMax > 10) {
+      secondaryMax = primaryMax * .75;
+    }
+    if (primaryMax > secondaryMax * 1.5) {
+      primaryMax = secondaryMax * 1.5;
+    }
+
+    primaryMax = Math.floor(primaryMax);
+    secondaryMax = Math.floor(secondaryMax);
+
+    let radiusFn;
+
+    if (this._size) {
+      const domain = extent(data, d => d.size);
+
+      if (domain[0] === domain[1]) {
+        domain[0] = 0;
+      }
+
+      radiusFn = scales.scaleLinear()
+        .domain(domain)
+        .rangeRound([3, min([primaryMax, secondaryMax])]);
+
+      const val = center.size;
+      center.r = radiusFn(val);
+    }
+    else {
+      radiusFn = scales.scaleLinear()
+        .domain([1, 2])
+        .rangeRound([primaryMax, secondaryMax]);
+    }
+
+    secondaries.forEach(s => {
+      s.ring = 2;
+      const val = this._size ? s.size : 2;
+      s.r = this._sizeMin ? max([this._sizeMin, radiusFn(val)]) : this._sizeMax ? min([this._sizeMax, radiusFn(val)]) : radiusFn(val);
+    });
+
+    primaries.forEach(p => {
+      p.ring = 1;
+      const val = this._size ? p.size : 1;
+      p.r = this._sizeMin ? max([this._sizeMin, radiusFn(val)]) : this._sizeMax ? min([this._sizeMax, radiusFn(val)]) : radiusFn(val);
+    });
+
+    nodes = [center].concat(primaries).concat(secondaries);
+
+    primaries.forEach(p => {
+      const check = ["source", "target"];
+      const {edge} = p;
+
+      check.forEach(node => {
+        edge[node] = nodes.find(n => n.id === edge[node].id);
+      });
+
+      edges.push(edge);
+
+      linkMap[p.id].forEach(edge => {
+        const node = edge.source.id === p.id ? edge.target : edge.source;
+
+        if (node.id !== center.id) {
+          let target = secondaries.find(s => s.id === node.id);
+
+          if (!target) {
+            target = primaries.find(s => s.id === node.id);
+          }
+
+          if (target) {
+            edge.spline = true;
+
+            const centerX = width / 2;
+            const centerY = height / 2;
+            const middleRing = (secondaryRing + primaryRing) / 1.75;
```

Comment:
did you try using `(secondaryRing + primaryRing) / 1.25` for one of the other handles? if not, let's look at it together.